wren-ai-service/src/providers/llm/openai.py

```diff
@@ -57,6 +59,26 @@ def __init__(
             base_url=api_base_url,
         )
 
+        # check if the model is actually Vertex AI model
+        self._vertexai_creds = None
+        if model.startswith("google/"):
+            self._vertexai_creds, _ = google.auth.default(
+                scopes=["https://www.googleapis.com/auth/cloud-platform"]
+            )
+
+    def __getattr__(self, name: str) -> Any:
+        # dealing with auto-refreshing expired credential
+        # https://cloud.google.com/vertex-ai/generative-ai/docs/multimodal/call-vertex-using-openai-library#refresh_your_credentials
+        if self._vertexai_creds and not self._vertexai_creds.valid:
+            auth_req = google.auth.transport.requests.Request()
+            self._vertexai_creds.refresh(auth_req)
+
+            if not self._vertexai_creds.valid:
+                raise RuntimeError("Unable to refresh auth")
+
+            self.client.api_key = self._vertexai_creds.token
+        return getattr(self.client, name)
+
```

Comment:
a bit curious, why did we validate the credit of vertexai in the openai provider?

Reply:
left comment